File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/SubstanceReferenceInformation.ts
import { CodeableConcept } from "./CodeableConcept";
import { Range } from "./Range";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type SubstanceReferenceInformationGene = BackboneElement & {
    gene?: CodeableConcept;
    source?: Reference[];
    geneSequenceOrigin?: CodeableConcept;
};

export type SubstanceReferenceInformationTarget = BackboneElement & {
    amount?: Quantity | string | Range;
    organism?: CodeableConcept;
    organismType?: CodeableConcept;
    amountType?: CodeableConcept;
    type?: CodeableConcept;
    interaction?: CodeableConcept;
    source?: Reference[];
    target?: Identifier;
};

export type SubstanceReferenceInformationGeneElement = BackboneElement & {
    type?: CodeableConcept;
    source?: Reference[];
    element?: Identifier;
};

export type SubstanceReferenceInformationClassification = BackboneElement & {
    domain?: CodeableConcept;
    source?: Reference[];
    subtype?: CodeableConcept[];
    classification?: CodeableConcept;
};

export type SubstanceReferenceInformation = DomainResource & {
    gene?: SubstanceReferenceInformationGene[];
    target?: SubstanceReferenceInformationTarget[];
    comment?: string;
    geneElement?: SubstanceReferenceInformationGeneElement[];
    classification?: SubstanceReferenceInformationClassification[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/SubstanceReferenceInformation.py
from typing import Optional, List
from pydantic import *
from ..base.CodeableConcept import CodeableConcept
from ..base.Range import Range
from ..base.Quantity import Quantity
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class SubstanceReferenceInformation_Gene(BackboneElement):
    gene: Optional[CodeableConcept] = None
    source: Optional[List[Reference]] = None
    gene_sequence_origin: Optional[CodeableConcept] = None

class SubstanceReferenceInformation_Target(BackboneElement):
    organism: Optional[CodeableConcept] = None
    organism_type: Optional[CodeableConcept] = None
    amount_type: Optional[CodeableConcept] = None
    type: Optional[CodeableConcept] = None
    interaction: Optional[CodeableConcept] = None
    source: Optional[List[Reference]] = None
    amount_quantity: Optional[Quantity] = None
    amount_string: Optional[str] = None
    target: Optional[Identifier] = None
    amount_range: Optional[Range] = None

class SubstanceReferenceInformation_GeneElement(BackboneElement):
    type: Optional[CodeableConcept] = None
    source: Optional[List[Reference]] = None
    element: Optional[Identifier] = None

class SubstanceReferenceInformation_Classification(BackboneElement):
    domain: Optional[CodeableConcept] = None
    source: Optional[List[Reference]] = None
    subtype: Optional[List[CodeableConcept]] = None
    classification: Optional[CodeableConcept] = None

class SubstanceReferenceInformation(DomainResource):
    gene: Optional[List[SubstanceReferenceInformation_Gene]] = None
    target: Optional[List[SubstanceReferenceInformation_Target]] = None
    comment: Optional[str] = None
    gene_element: Optional[List[SubstanceReferenceInformation_GeneElement]] = None
    classification: Optional[List[SubstanceReferenceInformation_Classification]] = None